gamemodes/terrortown/gamemode/shared/sh_player_ext.lua

```diff
@@ -992,44 +992,85 @@ function plymeta:SetSubRoleModel(mdl)
 	end
 end
 
+if SERVER then
+	util.AddNetworkString("TTT2ConfirmPlayer")
+end
+
+---
+-- Update player corpse state
+-- @param[opt] boolean announceRole
+-- @realm server
+function plymeta:ConfirmPlayer(announceRole)
+	if SERVER then
+		if self:GetNetworkingUInt("firstFound") <= 0 then
+			self:SetNetworkingUInt("firstFound", CurTime())
+		end
+
+		self:SetNetworkingUInt("lastFound", CurTime())
+
+		if announceRole then
+			self:SetNetworkingBool("roleFound", true)
+		end
+
+		self:SetNWBool("body_found", true) -- TODO just for compatibility
+		self:SetNetworkingBool("bodyFound", true)
+
+		net.Start("TTT2ConfirmPlayer")
```

Comment:
Better using syncing lists. Maybe adding callbacks (hook names?) to avoid exchanged network messages (just using one network message)

Reply:
I dont think that this is in the scope of this PR ;) as we did plan on reworking the confirmation (syncing) system anyway.